### PR TITLE
Share GPU ressources across render windows

### PR DESCRIPTION
**Pull request by bruyeret**
Add possibility to share a single context for several render windows.
This enables GPU ressource sharing accross render windows.

This feature uses `drawImage` from the main WebGL context to the child render window's canvas. It is slow if there are a lot of render windows on Firefox: see [this issue](https://bugzilla.mozilla.org/show_bug.cgi?id=1163426).

### CODE REVIEW

File: Sources/Rendering/OpenGL/RenderWindow/index.js [hunk 1]
```diff
@@ -1157,6 +1274,7 @@ const DEFAULT_VALUES = {
   shaderCache: null,
   initialized: false,
   context: null,
+  context2D: null,
```
[the remark this answers is not in the document]
Comment (jourdain) [reply Sources/Rendering/OpenGL/RenderWindow/index.js:1277]: we leave them commented out just so we know the expected set if keys.

File: Sources/Rendering/OpenGL/RenderWindow/index.js [hunk 2]
```diff
@@ -1206,30 +1325,27 @@ export function extend(publicAPI, model, initialValues = {}) {
   model._glInformation = null;
 
   model.myFactory = vtkOpenGLViewNodeFactory.newInstance();
-  /* eslint-disable no-use-before-define */
-  model.myFactory.registerOverride('vtkRenderWindow', newInstance);
-  /* eslint-enable no-use-before-define */
```
[the remark this answers is not in the document]
Comment (jourdain) [reply Sources/Rendering/OpenGL/RenderWindow/index.js:1211]: I have no idea what that is doing in the rendering scene graph. I won't be of much help.

**Comment by finetjul** [comment]
@sankhesh can you please review when you get a chance

**Review by finetjul — APPROVED** [review]
LGTM

**Comment by sankhesh** [comment]
@bruyeret Sorry, I missed the review period but the changes look good to me.  Excellent work!

**Pull request merged**